mobify/astro-scaffold
# [HYB 693] Welcome Modal Scaffold

Welcome Modal inside the scaffold

JIRA: [**[HYB-693] As a scaffold user, I want a welcome/signup screen (modal)**](https://mobify.atlassian.net/browse/HYB-693)
Linked PRs: **None**

## Changes
- Welcome Modal displays only once -- on initial startup if configured to do so
- Header is a feature switch that can be toggled in `welcomeConfig.js`
- Welcome Modal uses navigationPlugin if header is switched on
- Welcome Modal uses webViewPlugin if header is switched off

## Research
- http://maya.com/blog/on-touch-target-size

## How to test-drive this PR
- Run app and see welcome modal appear
- Check to see if the header can be toggled by going into `config/welcomeConfig.js` and switching `showHeader` property.
- On iOS simulator: `Simulator > Reset Contents & Settings...`
- Go to `app.js` and pass in `{forced: false}` into `welcomeModal.show()`
- Run app, make sure welcome modal shows up, close app.
- Run app on simulator again and make sure it doesn't show up.



## TODOS:
- ~~[ ] Tests are passing on your machine.~~
